home/tests/integration/views/web/test_data.py

```diff
@@ -0,0 +1,174 @@
+import csv
+import io
+
+from collections import defaultdict
+from datetime import date, datetime, timedelta
+from pytz import utc
+from typing import Dict, List
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.test import Client, TestCase
+
+from home.models import IntentionalWalk
+from home.utils.generators import (
+    AccountGenerator,
+    ContestGenerator,
+    DailyWalkGenerator,
+    DeviceGenerator,
+    IntentionalWalkGenerator,
+)
+
+
+class Login:
+    username = "testadmin"
+    password = "test*PW"
+
+    def __init__(self):
+        User.objects.create_user(username=self.username, password=self.password)
+
+
+class TestCsvViews(TestCase):
+    @staticmethod
+    def login(client: Client):
+        return client.login(username=Login.username, password=Login.password)
+
+    @classmethod
+    def setUpTestData(cls):
+        # Create user login
+        Login()
+
+        # Generate fake accounts
+        accounts = list(AccountGenerator().generate(2))
+        # Device associated with accounts[0]
+        device0 = list(DeviceGenerator(accounts[0:1]).generate(1))
+        # Device associated with accounts[1]
+        device1 = list(DeviceGenerator(accounts[1:2]).generate(1))
+
+        # Generate daily walks (10 per device)
+        dwalks0 = DailyWalkGenerator(device0)
+        dwalks1 = DailyWalkGenerator(device1)
+        for dt in range(10):
+            # Set dates on walks to 3000-03-01 to 3000-03-10
+            t = utc.localize(datetime(3000, 3, 1, 10, 0)) + timedelta(days=dt)
+            next(dwalks0.generate(1, date=t))
+            next(dwalks1.generate(1, date=t))
+
+        # Generate intentional walks (5, every other day)
+        iwalks0 = IntentionalWalkGenerator(device0)
+        iwalks1 = IntentionalWalkGenerator(device1)
+        for dt in range(5):
+            # Set dates on walks to [2, 4, 6, 8, 10] (3000-03)
+            t = utc.localize(datetime(3000, 3, 1, 10, 0)) + timedelta(days=(dt * 2))
+            next(iwalks0.generate(1, start=t, end=(t + timedelta(hours=2))))
+            next(iwalks1.generate(1, start=t, end=(t + timedelta(hours=2))))
+
+    @staticmethod
+    def date_from_timestamp(ts: str) -> datetime.date:
+        return datetime.fromisoformat(ts).date()
+
+    @staticmethod
+    def group_by(rows: List[Dict], key: str) -> Dict[str, List]:
+        grouped = defaultdict(list)
+        for row in rows:
+            grouped[row[key]].append(row)
+        return grouped
+
+    def test_user_agg_csv_view(self):
+        c = Client()
+        self.assertTrue(self.login(c))
+
+        params = {
+            "start": date(3000, 3, 7),
+            "end": date(3000, 3, 14),
+        }
+        contest_id = next(ContestGenerator().generate(1, **params)).pk
```

Comment:
`ContestGenerator::generate` creates a new contest in the db (a db that is launched just for tests) and returns the id (primary key) to `contest_id`.